### DISCUSSION TITLE
Coding standards question

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#5237] [Q&A] [answered]
Both String.Empty and string.Empty both compile but is one considered "more proper" than the other?  If so, which one?

**Comment by CyrusNajmabadi** [comment]
This question is out of scope for csharplang.   Questions about style are inherently up to you.  You can also use tools like style-cop, roslyn-analyzer, or roslyn itself to often specify which you want and have tools enforce that.  

For more questions like this, i recommend using one of the discord channels (like discord.gg/csharp).

**Reply by jmarolf** [reply]
@CyrusNajmabadi is correct that Microsoft has never been very opinionated about what the preferred style for C# code is. 


All that being said. The default editorconfig that is created with `dotnet new editorconfig` has these settings:
https://github.com/dotnet/templating/blob/0ea5134591d717f0da73c06ffc9b710c153bbc77/template_feed/Microsoft.DotNet.Common.ItemTemplates/content/EditorConfig/Dotnet/.editorconfig#L39-L40

These mean the style of `string.Empty` is preferred over `String.Empty`. That is and _extremely_ slight preference though.

**Comment by umbarov** [comment, marked as answer]
this post (by @jaredpar) might give additional info about `string` and `String` https://blog.paranoidcoding.com/2019/04/08/string-vs-String-is-not-about-style.html
